app/controllers/sign_up/completions_controller.rb

```diff
@@ -0,0 +1,28 @@
+module SignUp
+  class CompletionsController < ApplicationController
+    def show
+      if user_fully_authenticated? && !session[:sp].nil?
```

Comment:
sure. I initially did not go that route because `{}.present?` returns false, but if we had the `sp` session var set to `{}`, I would consider that having a service provider present.

The way our code currently works, it would never be set to `{}`, it would either be a hash with keys and values or `nil`. BUT that could change. Are we OK with checking for `present?` even if that means returning false if the `session[:sp]` is equal to an empty hash?